extensions/amp-base-carousel/1.0/amp-base-carousel.js

```diff
@@ -0,0 +1,140 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {ActionTrust} from '../../../src/action-constants';
+import {BaseCarousel} from './base-carousel';
+import {CSS} from '../../../build/amp-base-carousel-1.0.css';
+import {PreactBaseElement} from '../../../src/preact/base-element';
+import {Services} from '../../../src/services';
+import {createCustomEvent} from '../../../src/event-helper';
+import {dict} from '../../../src/utils/object';
+import {isExperimentOn} from '../../../src/experiments';
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {userAssert} from '../../../src/log';
+
+/** @const {string} */
+const TAG = 'amp-base-carousel';
+
+/**
+ * Triggers a 'slideChange' event with one data param:
+ * 'index' - index of the current slide.
+ * @param {!Window} win
+ * @param {!../../../src/service/action-impl.ActionService} action
+ * @param {!Element} el The element that was selected or deslected.
+ * @param {number} index
+ * @param {!ActionTrust} trust
+ * @private
+ */
+function fireSlideChangeEvent(win, action, el, index, trust) {
+  const name = 'slideChange';
+  const slideChangeEvent = createCustomEvent(
+    win,
+    `amp-base-carousel.${name}`,
+    dict({'index': index})
+  );
+  action.trigger(el, name, slideChangeEvent, trust);
+}
+
+class AmpBaseCarousel extends PreactBaseElement {
+  /** @override */
+  init() {
+    const {element} = this;
+    const action = Services.actionServiceForDoc(element);
```

Comment:
Can we resolve this in the `fireSlideChangeEvent`?

Reply:
Done.